### PR TITLE
Ts SimpleTable

### PR DESCRIPTION
**Pull request by gustavo-salazar**
Migration to typescript and visual framework for the `SimpleTable` and subcomponents.

Components Migrated:
* `Table/Body`
* `Table/Row`
* `Table/Column`
* `Table/Views/Table`
* `Table/SimpleTable`

`SimpleTable` is currently only used for `KeySpeciesTable` as in [this page in the live site](https://www.ebi.ac.uk/interpro/entry/InterPro/IPR000001/taxonomy/uniprot/#keyspecies). So this PR also migrates `KeySpeciesTable` to ha ve typescript example case for it.

Other changes:
* Adding specificity to some `EntryMenuLink` styles so it doesn't change with some removed foundation imports.
* Moving table styles form `interpro-new.css` into `Table/styles.css.

**Note:** Second of a series of PRs chained to migrate `<Table>` and all its subcomponents. Make sure to review #626 first. And, when merging, delete the branch to update the next PRs.

### CODE REVIEW

**Review by tgrego — COMMENTED** [review]
The KeySpeciesTable in the example given seems to have the last column misaligned with the listed icons (Looks like the header used to be centered as is now too left aligned).

**Review by tgrego — APPROVED** [review]
All looks good

File: src/components/Table/views/Table/index.tsx [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+const css = cssBinder(styles, fonts);
+
+type Props<
+  RowData = Record<string, unknown>,
+  ExtraData = Record<string, unknown>,
+> = {
+  dataTable: Array<RowData & {}>;
+  rowKey?: string;
+  isStale?: boolean;
```
Comment (matthiasblum) [inline src/components/Table/views/Table/index.tsx:20]: Remove `isStale` as it's never used?

```suggestion
```

File: src/components/Table/views/Table/index.tsx [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        data-testid="data-table"
+      >
+        <Header columns={columns} notFound={notFound} />
+        <Body<RowData, ExtraData>
+          rows={dataTable || []}
+          rowKey={rowKey}
+          columns={columns}
+          notFound={!!notFound}
+          loading={!!loading}
+          // isStale={!!isStale}
```
Comment (matthiasblum) [inline src/components/Table/views/Table/index.tsx:61]:
```suggestion
```

File: src/components/Table/views/Table/index.tsx [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+};
+
+class TableView<
+  RowData = Record<string, unknown>,
+  ExtraData = Record<string, unknown>,
+> extends PureComponent<Props<RowData, ExtraData>> {
+  render() {
+    const {
+      loading,
+      // isStale,
```
Comment (matthiasblum) [inline src/components/Table/views/Table/index.tsx:38]:
```suggestion
```

**Review by matthiasblum — APPROVED** [review]
LGTM.

I have two suggestions, but there are not in the scope of this PR, it's just I noticed it while reviewing it:

- `no-decoration` should be applied to the link to open the list of proteins https://github.com/ProteinsWebTeam/interpro7-client/blob/a874fc8ecd90ca25e0475611af6f5b0f16ee6569/src/components/Matches/ProteinDownloadRenderer/index.tsx#L33
- The button to dowload fasta sequences should be the `icon-FASTA`

If you can make the changes in this PR, great. Otherwise it'll be for a future PR.

File: src/components/Table/style.css [hunk 4]
```diff
@@ -1,6 +1,181 @@
...
+  border: 0;
+  background: transparent; /*Overwrite EBI global*/
+}
+table.light thead th {
+  vertical-align: top;
+  background: rgba(255, 255, 255, 0.8);
+  outline: none;
+  border-bottom: 1px solid #d4dddf;
+  text-transform: uppercase;
+  font-size: 0.93em;
```
Comment (matthiasblum) [inline src/components/Table/style.css:63]:
```suggestion
```

File: src/components/Table/style.css [hunk 5]
```diff
@@ -1,6 +1,181 @@
...
+
+/*Style light table */
+table.light {
+  color: var(--colors-light-txt);
+  font-size: 0.93em;
+  border-collapse: collapse;
+}
+table.light thead {
+  color: #9baaaf;
+  font-size: 0.9em;
```
Comment (matthiasblum) [inline src/components/Table/style.css:52]:
```suggestion
```

**Review by matthiasblum — APPROVED** [review]

**Comment by gustavo-salazar** [comment]
> * `no-decoration` should be applied to the link to open the list of proteins https://github.com/ProteinsWebTeam/interpro7-client/blob/a874fc8ecd90ca25e0475611af6f5b0f16ee6569/src/components/Matches/ProteinDownloadRenderer/index.tsx#L33

I did this one!

> * The button to dowload fasta sequences should be the `icon-FASTA`

But this one requires changes in multiple files to pass the custom Icon down to the component that actually renders it. will open an issue from it to work on it later.

**Pull request merged**